api/service/projects_service.go

```diff
@@ -5,22 +5,31 @@ import (
 	"fmt"
 	"strings"
 
+	"bytes"
+	"encoding/json"
+	"html/template"
+	"net/http"
+
 	"golang.org/x/exp/slices"
 
 	"github.com/pkg/errors"
 
 	"github.com/caraml-dev/mlp/api/repository"
 
+	"github.com/caraml-dev/mlp/api/config"
 	"github.com/caraml-dev/mlp/api/models"
 	"github.com/caraml-dev/mlp/api/pkg/authz/enforcer"
 )
 
 type ProjectsService interface {
 	ListProjects(ctx context.Context, name string, user string) ([]*models.Project, error)
 	CreateProject(ctx context.Context, project *models.Project) (*models.Project, error)
-	UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error)
+	UpdateProject(ctx context.Context, project *models.Project) (*models.Project, string, error)
```

Comment:
can we don't change the interface for this? its off-sync with the rest of the method, instead let the controller call `UpdateProject` then call this as a singular separate method?

Reply:
after thinking can we move all the additional logic to projects_api for now? the service layer shouldn't be dealing with how the http responds are formed